Epsilon/Component/KpiMatrixComponent.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Epsilon.Abstractions.Component;
+using Epsilon.Abstractions.Model;
+using Epsilon.Canvas;
+using Epsilon.Canvas.Abstractions.QueryResponse;
+using Epsilon.Canvas.Abstractions.Service;
+using Microsoft.Extensions.Configuration;
+
+namespace Epsilon.Component;
+
+public class KpiMatrixComponent : Component<KpiMatrix>
+{
+    
+   
+    private readonly IConfiguration _configuration;
+    private readonly IGraphQlHttpService _graphQlService;
+    private readonly IAccountHttpService _accountHttpService;
+
+    public KpiMatrixComponent(
+        IGraphQlHttpService graphQlService,
+        IAccountHttpService accountHttpService,
+        IConfiguration configuration
+    )
+    {
+        _graphQlService = graphQlService;
+        _accountHttpService = accountHttpService;
+        _configuration = configuration;
+    }
+    
+    public async override Task<KpiMatrix> Fetch()
+    {
+        var courseId = _configuration["Canvas:courseId"];
+
+        var outcomesQuery = QueryConstants.GetUserKpiMatrixOutcomes.Replace("$courseId", $"{courseId}");
+        var outcomes = await _graphQlService.Query<GetUserKpiMatrixOutcomes>(outcomesQuery);
+        
+        var kpiMatrix = ConvertToComponent(outcomes);
+
+        return kpiMatrix;
+    }
+
+    private GradeStatus GetGradeStatus(double? points)
+    {
+        return points switch
+        {
+            null => GradeStatus.NotGraded,
+            >= 4 => GradeStatus.Approved,
```

Comment:
3 is already an approved grade. 
Ideally, we need to look at the "mastery" point.  And not a value that we define in our code.